Session bar accessibility fixes

foxbunny opened the pull request:
- Use toggle tip for the public/private/network badge
- Create a new popup component for the tz selector and profile options

Also fixes #6757.

# Before

<img width="757" height="945" alt="2025-10-07 07_05_59-Screenshots - File Explorer" src="https://github.com/user-attachments/assets/0e4da60e-ec9e-42ae-af04-9773a609e910" />

# After

<img width="757" height="945" alt="2025-10-07 07_05_69-Screenshots - File Explorer" src="https://github.com/user-attachments/assets/729ecda9-cecb-4366-bfc3-939247933399" />

# WCAG criteria

- 1.4.3 Contrast (Minimum) (Level AA)
- 1.3.1 Info and Relationships (Level A)
- 1.4.13 Content on Hover or Focus (Level AA)
- 2.4.7 Focus Visible (Level AA)

foxbunny committed:
Session bar accessibility fixes

- Use toggle tip for the public/private/network badge
- Create a new popup component for the tz selector and profile options

ThiefMaster commented:
Looks fine in general, but it has the same issue that we had in a few other places: Clicking the trigger again does not close the tooltip but rather makes it disappear for a fraction of a second and then come back.

ThiefMaster commented:
Actually, some text in the timezone picker looks really strange now. Almost like broken antialiasing...

![image](https://github.com/user-attachments/assets/1688b87c-e074-4af7-b07b-6122ad65b351)

foxbunny commented:
> Actually, some text in the timezone picker looks really strange now. Almost like broken antialiasing...

What browser/OS are you testing on?

foxbunny commented:
> Clicking the trigger again does not close the tooltip but rather makes it disappear for a fraction of a second and then come back.

As I explained before, this is the expected behavior of a toggle tip. A toggle tip is a hidden piece of information that is shown when an associated button is clicked. It shows. It doesn't toggle (even though the name would suggest it does). This is how it's announced to the user and that's why it has no indicator of the "open" state.

I could technically implement it the way you want but it would break the expectation for users relying on a11y cues and also introduce unnecessary noise about the change in state.

Let me know if you insist on making toggle tips actually toggle on/off.

ThiefMaster commented:
> What browser/OS are you testing on?

Firefox on Windows 10

> Let me know if you insist on making toggle tips actually toggle on/off.

It seems to have no drawbacks to close on a second interaction while it's open (even if it's not "announced" to users), so I'd prefer the toggle behavior.

(FWIW, staying open on a repeated click would probably not be as annoying if it actually stayed open and did not "reopen" while already open.)

tomasr8 commented:
I noticed a weird behaviour with the visibility tooltip. Sometimes when you click on it, the tooltip 'triangle' seems to somehow move from the top of the page into its intended position. I'm not sure if it's just Firefox but it looks a bit strange. Here's a video, it happens on the second click:

https://github.com/user-attachments/assets/173511b9-74cf-4df2-92f7-b277719cb119

foxbunny commented:
Addressed issues with toggle tip and its arrow. 

Still no solution for the Firefox (that is, I cannot reproduce on my end). I see the font rendering is not great on Windows 10 with Firefox, but that is across the page on BrowserStack (not popup-specific), so I'm not sure if that's the issue you're seeing @ThiefMaster. I've seen reports of various rendering issues with Roboto on Firefox and Chrome on both Windows and MacOS, so I suspect is an issue with this particular font in combination with various quirks of different font rendering engines, but, again, this is just speculation. It renders fine on a physical Windows 11 machine in all browsers I can test on.

I've noticed that bumping the font size to 14px makes it render slightly better (currently base size is 13px 😱), but I can't be sure if this fixes the issue without breaking it in some other browser.

ThiefMaster commented:
Another strange behavior I found: When opening the protection info box, and then (while it's open) one of the others, it seems to be pretty random whether the protection tooltip closes or not. It happens frequently enough that reproducing it should be very easy:

![image](https://github.com/user-attachments/assets/c8f147fd-d495-4a62-9556-70403b4a8c16)

ThiefMaster commented:
![image](https://github.com/user-attachments/assets/a59cded3-0b8b-4e46-9491-3eb490bf886d)

The rounded/regular borders don't seem to be correct. Hard to notice at normal zoom, but whne zoomng in it's very obvious

ThiefMaster commented:
![image](https://github.com/user-attachments/assets/5b80f0bf-977e-4655-9c18-e82ab33ec618)

There's also some spacing between the arrow and the popup box

GovernmentPlates commented on indico/web/templates/_session_bar.html:
```diff
@@ -6,169 +6,182 @@
 
-        <button id="tz-selector-link" class="i-button icon-time arrow">{{ timezone_data.active_tz }}</button>
-    {% endif %}
+                        <div id="tz-profile-setting">
+                            <span>{% trans %}Your profile timezone:{% endtrans %}</span>
+                            <span>{{ profile_timezone }}</span>
+                        </div>
```
Something I noticed: The `{{ profile_timezone }}` is empty for users who aren't logged-in (which is fine), but it does make things look a bit weird (see below).

![image](https://github.com/user-attachments/assets/7528f05e-32b2-4ff0-8c20-82cfe0a35215)

Maybe we could just hide this section if there isn't a profile timezone?

foxbunny replied:
Fixed

ThiefMaster commented:
Could you fix the rebase conflicts in indico/web/templates/_session_bar.html?

ThiefMaster commented:
Not sure if it's fixed by something else in master or another PR, but right now the session bar in a meeting event when you're logged in (the one with "public" or "protected") has a short flash of the tooltip opening on the top of the page and then only repositioning it a moment later.

It only happens for the protection one, timezone/user seem fine.

foxbunny commented:
Hey, are we going to do something about this PR?

ThiefMaster commented:
Happy to revisit this one, but it would need to be rebased, there are quite a few conflicts... :/

ThiefMaster commented on indico/web/client/js/custom_elements/_base.js:
```diff
@@ -161,19 +161,26 @@ export default class CustomElementBase extends HTMLElement {
     this.setup?.();
     this.setup = null;
     this.dispatchEvent(new Event('x-connect'));
+    this.onconnect?.();
```
Is this custom? If yes, let's rename it to `onConnect`, the current capitalization is weird...

foxbunny replied:
It's custom, and it's following the convention of the DOM API.

ThiefMaster replied:
ok, fine then

ThiefMaster commented on indico/web/client/js/widgets/lang_selector.js:
```diff
@@ -0,0 +1,11 @@
+// LICENSE file for more details.
+
+// eslint-disable-next-line import/unambiguous
+window.addEventListener('DOMContentLoaded', () => {
```
I think you added `domReady` for this not too long ago ;)

ThiefMaster commented on indico/web/client/js/widgets/lang_selector.js:
```diff
@@ -0,0 +1,11 @@
+
+// eslint-disable-next-line import/unambiguous
+window.addEventListener('DOMContentLoaded', () => {
+  document.getElementById('language-selector-inline').onclick = evt => evt.preventDefault();
```
I'm not a fan of setting `onclick` directly instead of using `addEventListener`...

I think it would also be a good to add a comment here why this is needed.

ThiefMaster commented on indico/web/client/styles/design_system/variables.scss:
```diff
@@ -39,6 +39,22 @@ body {
+  --surface-yellow-color: #ffba0c;
+  --surface-yellow-secondary-color: #ffbc37;
+  --text-on-yellow-color: #000;
+  --surface-organge-color: #f80;
+  --surface-organge-secondary-color: #ae6512;
```
s/organge/orange/

ThiefMaster commented on indico/web/templates/_session_bar.html:
```diff
@@ -6,175 +6,195 @@
-    </div>
-
-    <button id="protection-details-link"
-       class="arrow i-button protection-{{ mode }} {{ 'icon-shield' if mode != 'public' else 'icon-earth' }}">
```
Did you remove the earth icon for public content on purpose?

foxbunny replied:
Yes and no. I was going to re-add it via CSS then it slipped my mind.

ThiefMaster commented on indico/web/templates/_session_bar.html:
```diff
@@ -6,175 +6,195 @@
+            <dialog data-dialog>
+                <article id="tz-selector-widget" aria-labelledby="tz-selector">
+                    <div>
+                        <h2 id="tz-selector">{% trans %}Choose timezone{% endtrans %}</h2>
```
The bold still looks quite bad there. Using `--text-color` for it instead of hard black makes it much nicer.

Likewise for the labels such as "Your profile timezone", "Use timezone based on:" and "Remember these settings"

tomasr8 commented on indico/web/client/styles/partials/_main.scss:
```diff
@@ -222,36 +225,117 @@ nav.main-breadcrumb {
+  flex: none;
+  display: flex;
+  align-items: center;
+  gap: 0.1em;
```
Not a big fan of the gap between the buttons to be honest. Could we either remove it, or use some subtle separator?

<img width="343" height="62" alt="Image" src="https://github.com/user-attachments/assets/f0655c5f-102a-4e8d-b31b-1ae5d17130a6" />

<img width="345" height="50" alt="Image" src="https://github.com/user-attachments/assets/912f4558-7608-499e-b290-9a215d95ae4d" />

tomasr8 commented on indico/web/client/styles/partials/_main.scss:
```diff
@@ -222,36 +225,117 @@ nav.main-breadcrumb {
+  align-items: center;
+  gap: 0.1em;
 
-  .toolbar {
```
When you activate the language dropdown, there is this white triangle that partially covers the `M` in "My language":

<img width="412" height="743" alt="Image" src="https://github.com/user-attachments/assets/4d43bf60-6235-40f3-a64c-6c8429cce420" />

foxbunny committed:
Various fixes

- Fix positioning issues with misaligned arrows
- Remove gaps between session bar buttons
- Use domReady instead of DOMContentLoaded event
- Use --text-color instead of pure black
- Remove bold style from headings and labels

foxbunny committed:
Update copyright headers
